### PR TITLE
Fix metadata drifting on workbench instances

### PR DESCRIPTION
**Pull request by bcreddy-gcp**
fixes https://github.com/hashicorp/terraform-provider-google/issues/18827

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
workbench: fixed a bug with workbench instance metadata drifting when using custom containers.
```

### CODE REVIEW

File: mmv1/templates/terraform/constants/workbench_instance.go [hunk 1]
```diff
@@ -27,6 +27,7 @@ var WorkbenchInstanceProvidedMetadata = []string{
     "agent-health-check-interval-seconds",
     "agent-health-check-path",
     "container",
+	"cos-update-strategy",
```
Comment (c2thorn) [inline mmv1/templates/terraform/constants/workbench_instance.go:30]:
```suggestion
    "cos-update-strategy",
```

File: mmv1/templates/terraform/constants/workbench_instance.go [hunk 2]
```diff
@@ -45,6 +46,7 @@ var WorkbenchInstanceProvidedMetadata = []string{
     "generate-diagnostics-bucket",
     "generate-diagnostics-file",
     "generate-diagnostics-options",
+	"google-logging-enabled",
```
Comment (c2thorn) [inline mmv1/templates/terraform/constants/workbench_instance.go:49]:
```suggestion
    "google-logging-enabled",
```

**Review by c2thorn — CHANGES_REQUESTED** [review]
is there an associated github issue and/or bug? Is this scenario covered by tests / how was this tested?

**Comment by bcreddy-gcp** [comment]
b/357693909
https://github.com/hashicorp/terraform-provider-google/issues/18827.

**Comment by bcreddy-gcp** [comment]
Similar PR: https://github.com/GoogleCloudPlatform/magic-modules/pull/10900

File: mmv1/templates/terraform/constants/workbench_instance.go [hunk 3]
```diff
@@ -67,6 +69,7 @@ var WorkbenchInstanceProvidedMetadata = []string{
     "report-system-status",
     "restriction",
     "serial-port-logging-enable",
+	"service-account-mode",
```
Comment (c2thorn) [inline mmv1/templates/terraform/constants/workbench_instance.go:72]:
```suggestion
    "service-account-mode",
```

**Review by c2thorn — APPROVED** [review]

**Review by c2thorn — APPROVED** [review]

**Pull request merged**